### ISSUE TITLE
How to Save Retrieved Documents in Retrieval Tasks

### ISSUE THREAD

**Issue by AugustLHHHHHH** [beir-cellar/beir#171]
I am currently working on an information retrieval task and I want to save the retrieved documents for bad case analysis.
I have printed the output of the search function in beir/retrieval/search/dense/exact_search.py, and the format is {q_id: {pid: score}}. such as:

![image](https://github.com/beir-cellar/beir/assets/109325850/9b4b1d23-4d5a-4080-86bc-9e99794b05f9)

However, it seems that the corpus ids are not sorted according to the scores. Should I sort the results by cosine similarity from high to low and save them so that the most relevant documents are placed at the beginning?

Thanks!

**Comment by thakur-nandan** [comment]
yes, the scores are not sorted, you can use this line to sort them:
```python
for query_id in results:
    scores_sorted = sorted(results[query_id].items(), key=lambda item: item[1], reverse=True)
```
